File: features/notice/notice/Notice.go
package notice

import (
	"github.com/hailongz/golang/db"
)

type Notice struct {
	db.Object
	Type	int32	`json:"type" name:"type" title:"通知类型" index:"ASC"`
	Fid	int64	`json:"fid" name:"fid" title:"消息来源ID" index:"ASC"`
	Iid	int64	`json:"iid" name:"iid" title:"消息来源项ID" index:"ASC"`
	Uid	int64	`json:"uid" name:"uid" title:"用户ID" index:"ASC"`
	Body	string	`json:"body" name:"body" title:"通知内容" length:"-1"`
	Options	interface{}	`json:"options,omitempty" name:"options" title:"其他数据" length:"-1"`
	Ctime	int64	`json:"ctime" name:"ctime" title:"创建时间"`
}

func (O *Notice) GetName() string {
	return "notice"
}

func (O *Notice) GetTitle() string {
	return "通知"
}

File: features/notice/ts/Notice.ts
import { int64, int32 } from "./lib/less";

/**
 * 通知
 * @type db
 */
export class Notice {

    /**
     * ID
     */
    id: int64 = 0

    /**
     * 通知类型
     * @index ASC
     */
    type: int32 = 0

    /**
     * 消息来源ID
     * @index ASC
     */
    fid: int64 = 0

    /**
     * 消息来源项ID
     * @index ASC
     */
    iid: int64 = 0

    /**
     * 用户ID
     * @index ASC
     */
    uid: int64 = 0

    /**
     * 通知内容
     * @length -1
     */
    body: string = ""

    /**
     * 其他数据
     * @length -1
     */
    options: any

    /**
     * 创建时间
     */
    ctime: int64 = 0

}